tests/examples/wallet/test_wallet.py

```diff
@@ -1,73 +1,106 @@
     # Approve fails if not enough value is sent
-    assert_tx_failed(lambda: x.approve(1, to, value, data, [sign(1, to, value, data, k) if k else [0, 0, 0] for k in (t.k1, 0, t.k3, 0, t.k5)], value=0, sender=t.k1))
-    assert x.approve(1, to, value, data, [sign(1, to, value, data, k) if k else [0, 0, 0] for k in (t.k1, 0, t.k3, 0, t.k5)], value=value, sender=t.k1)
+    sigs = pack_and_sign(1, k1, 0, k3, 0, k5)
+    assert_tx_failed(lambda: c.approve(1, to_address, value, data, sigs, transact={'value': 0, 'from': a1}))
+    ssigs = pack_and_sign(1, k1, 0, k3, 0, k5)
+    assert c.approve(1, to_address, value, data, sigs, call={'value': value, 'from': a1})
+
     print("Basic tests passed")
 
 
-def test_javascript_signatures():
+def test_javascript_signatures(w3, get_contract):
+    a3 = w3.eth.accounts[2]
     # The zero address will cause `approve` to default to valid signatures
     zero_address = "0x0000000000000000000000000000000000000000"
-    accounts = ["0x776ba14735ff84789320718cf0aa43e91f7a8ce1", "0x095ce4e4240fa66ff90282c26847456e3f3b5002"]
+    accounts = [
+        "0x776ba14735ff84789320718cf0aa43e91f7a8ce1",
+        "0x095ce4e4240fa66ff90282c26847456e3f3b5002"
+    ]
     # The address that will receive the transaction
-    recipient = "0x776ba14735ff84789320718cf0aa43e91f7a8ce1"
+    recipient = "0x776Ba14735FF84789320718cf0aa43e91F7A8Ce1"
     # These are the matching sigs to the accounts
     raw_sigs = [
         "0x4a89507bf71749fb338ed13fba623a683d9ecab0fb9c389a4298525c043e38281a00ab65628bb18a382eb8c8b4fb4dae95ccc993cf49f617c60d8051180778601c",
         "0xc84fe5d2a600e033930e0cf73f26e78f4c65b134f9c9992f60f08ce0863abdbe0548a6e8aa2d952659f29c67106b59fdfcd64d67df03c1df620c70c85578ae701b"
     ]
 
     # Turns the raw sigs into sigs
-    sigs = [(utils.big_endian_to_int(x[64:]), utils.big_endian_to_int(x[:32]), utils.big_endian_to_int(x[32:64])) for x in
-            map(lambda z: utils.decode_hex(z[2:]), raw_sigs)]
+    sigs = [
+        (
+            w3.toInt(x[64:]),  # v
+            w3.toInt(x[:32]),  # r
+            w3.toInt(x[32:64])  # s
+        ) for x in map(lambda z: w3.toBytes(hexstr=z[2:]), raw_sigs)
+    ]
+
+    h = w3.sha3((0).to_bytes(32, "big") + b'\x00' * 12 + w3.toBytes(hexstr=recipient[2:]) + (25).to_bytes(32, "big") + b'')
+    h2 = w3.sha3(b"\x19Ethereum Signed Message:\n32" + h)
 
-    h = utils.sha3(utils.encode_int32(0) + b'\x00' * 12 + utils.decode_hex(recipient[2:]) + utils.encode_int32(25) + b'')
-    h2 = utils.sha3(b"\x19Ethereum Signed Message:\n32" + h)
     # Check to make sure the signatures are valid
-    assert '0x' + utils.encode_hex(utils.sha3(utils.ecrecover_to_pub(h2, sigs[0][0], sigs[0][1], sigs[0][2]))[12:]) == accounts[0]
-    assert '0x' + utils.encode_hex(utils.sha3(utils.ecrecover_to_pub(h2, sigs[1][0], sigs[1][1], sigs[1][2]))[12:]) == accounts[1]
+    assert Account.recoverHash(h2, sigs[0]).lower() == accounts[0]
```

Comment:
If you want, you can use `eth_utils.is_same_address` which handles equality for addresses across checksum, normal hex, and bytes representations.